### PR TITLE
adds ability to have scrollable back to top

### PR DESCRIPTION
**Pull request by N6REJ**
Pull Request for Issue # .

### Summary of Changes
Adds ability for "smart" back to top where...
1) back to top = on & scroll = off  : button is in footer within container.
2) back to top = off & scroll ( doesn't matter )  : no button displayed
3) back to top = on & scroll = on : button moved out of container and scrolls with viewport.
button is hidden until content is scrolled.

### Testing Instructions
check that backtotop button works in footer and turns on/off
load pr
run npm ci
check that button states are as listed above.

### Expected result
**backend**
image

**desktop**
image

**iphone11**
image

**note8**
image
image

### Documentation Changes Required
none

### CODE REVIEW

**Comment by brianteeman** [comment]
what is the difference between back to top and scroll to top?

**Comment by N6REJ** [comment]
guess I should include backend screen shot... back to top is stuck in footer

**Comment by brianteeman** [comment]
I meant what is the difference so that we need two

**Comment by N6REJ** [comment]
if they choose back-to-top its a sticky that is ONLY in the footer container.  Never leaves footer.
if they choose "scroll" then it scrolls with page and is not part of the footer itself.

**Comment by brianteeman** [comment]
Thank you for the explanation  - my 2c I would only offer one

**Comment by N6REJ** [comment]
> 
> 
> Thank you for the explanation - my 2c I would only offer one

I agree but having a back-to-top that doesn't stay in the viewport is very non traditional and what they wanted as default state.

**Comment by N6REJ** [comment]
I wish I could hide when above the fold but w/o javascript thats not possible afaik

**Comment by brianteeman** [comment]
You can do it with just css https://moderncss.dev/pure-css-smooth-scroll-back-to-top/

**Comment by richard67** [comment]
> Thank you for the explanation - my 2c I would only offer one

100% agree. It makes no sense to have both of them, and it may even confuse the user.

File: templates/cassiopeia/scss/blocks/_footer.scss [hunk 1]
```diff
@@ -8,11 +8,11 @@
 
   .grid-child {
     padding: 2.5rem ($cassiopeia-grid-gutter/2);
-  }
+    }
```
Comment (richard67) [inline templates/cassiopeia/scss/blocks/_footer.scss]:
```suggestion
  }
```
Why are you doing unrelated and wrong code style changes?

File: templates/cassiopeia/scss/blocks/_footer.scss [hunk 2]
```diff
@@ -8,11 +8,11 @@
...
 
   a {
     color: currentColor;
-  }
+    }
```
Comment (richard67) [inline templates/cassiopeia/scss/blocks/_footer.scss]:
```suggestion
  }
```
Why are you doing unrelated and wrong code style changes?

File: templates/cassiopeia/scss/blocks/_footer.scss [hunk 3]
```diff
@@ -22,14 +22,36 @@
     color: var(--cassiopeia-color-primary);
     background: $white;
     border-radius: $border-radius;
-  }
+    }
```
Comment (richard67) [inline templates/cassiopeia/scss/blocks/_footer.scss]:
```suggestion
  }
```
Why are you doing unrelated and wrong code style changes?

File: templates/cassiopeia/scss/blocks/_footer.scss [hunk 4]
```diff
@@ -22,14 +22,36 @@
...
 
   [dir=rtl] & {
     background-image: $cassiopeia-header-grad-rtl;
 
     .back-top {
       margin-right: auto;
       margin-left: 0;
+      }
+    }
```
Comment (richard67) [inline templates/cassiopeia/scss/blocks/_footer.scss]:
```suggestion
    }
  }
```
Fix indentation.

File: templates/cassiopeia/scss/blocks/_footer.scss [hunk 5]
```diff
@@ -22,14 +22,36 @@
...
+    background: $white;
+    border-radius: $border-radius;
+    cursor: pointer;
+    text-decoration: none;
+
+    &-top {
+      right: 1rem;
+      }
     }
   }
```
Comment (richard67) [inline templates/cassiopeia/scss/blocks/_footer.scss]:
```suggestion
    }
  }
}
```
Fix indentation.

**Comment by brianteeman** [comment]
please fix the code style issues with the last commit

image

File: templates/cassiopeia/scss/blocks/_footer.scss [hunk 6]
```diff
@@ -52,4 +52,31 @@
       right: 1rem;
     }
   }
+   */
```
Comment (richard67) [inline templates/cassiopeia/scss/blocks/_footer.scss]: Please remove commented out code.

Comment (N6REJ) [reply templates/cassiopeia/scss/blocks/_footer.scss]: done in last commit

**Comment by N6REJ** [comment]
@brianteeman @richard67 thanks guys, idk how that stuff slipped past me.

**Comment by hans2103** [comment]
@N6REJ 
Back-to-top Link = YES
Scroll-to-top = YES

When I scroll to the bottom of the page I see the following styling:

Schermafbeelding 2020-10-05 om 17 00 49

File: templates/cassiopeia/scss/blocks/_global.scss [hunk 7]
```diff
@@ -89,3 +89,13 @@ meter {
     color: var(--cassiopeia-color-link);
   }
 }
+
+.awesomplete {
+  input {
+    @include border-right-radius(0);
+  }
+}
```
Comment (hans2103) [inline templates/cassiopeia/scss/blocks/_global.scss:97]: @N6REJ What has awesomeplete to do with this PR?

**Comment by hans2103** [comment]
@N6REJ I've tried to create a PR on yours, but failed. Therefor I have created a new PR https://github.com/joomla/cassiopeia/pull/169

**Comment by N6REJ** [comment]
@hans2103 I'll just let you run with it then... You've changed it quite a bit.

File: templates/cassiopeia/scss/blocks/_global.scss (hunk 7 above)
Comment (N6REJ) [reply templates/cassiopeia/scss/blocks/_global.scss:97]: pretty sure I didn't put it there

Comment (N6REJ) [reply templates/cassiopeia/scss/blocks/_global.scss:97]: regarding the scroll it wasn't that way before.  I believe it's due to the changes you made.  I spent quite a bit of time getting the block uniform.

**Pull request closed without merging**